edgelet/edgelet-docker/src/runtime.rs

```diff
@@ -590,6 +596,91 @@ impl ModuleRuntime for DockerModuleRuntime {
         )
     }
 
+    fn system_resources(&self) -> Self::SystemResourcesFuture {
+        info!("Querying system resources...");
+        let mut system_info = sysinfo::System::new();
+        system_info.refresh_all();
+
+        let client = self.client.clone();
+        let docker_stats = self
+            .list() // Get all modules
+            .and_then(|modules: Vec<Self::Module>| {
+                // Get iterable of stats
+                remove_not_found(
+                    stream::iter_ok(modules)
+                        .and_then(move |module| {
+                            client.container_api().container_stats(module.name(), false)
+                        })
+                        .map_err(|err| {
+                            Error::from_docker_error(
+                                err,
+                                ErrorKind::RuntimeOperation(RuntimeOperation::SystemResources),
+                            )
+                        }),
+                )
+                .collect()
+            })
+            .map(serde_json::Value::Array) // Condense into single json value
+            .and_then(|stats| {
+                // convert to string
+                serde_json::to_string(&stats).map_err(|_| {
+                    Error::from(ErrorKind::RuntimeOperation(
+                        RuntimeOperation::SystemResources,
+                    ))
+                })
+            });
+
+        let uptime: u64 = uptime_lib::get()
+            .map(|u| u.num_seconds())
+            .unwrap_or_default()
+            .try_into()
+            .unwrap_or_default();
+
+        let current_time = SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .unwrap()
```

Comment:
`Err` here is unlikely, but please handle it instead of `unwrap()`ping: `.duration_since(...).map(Duration::as_secs).unwrap_or_default()` so that `current_time` just becomes 0 instead.